crates/bevy_asset/src/handle.rs

```diff
@@ -149,26 +155,34 @@ impl<T: Asset> Handle<T> {
         }
     }
 
-    /// Get a copy of this handle as a Weak handle
-    pub fn as_weak<U: Asset>(&self) -> Handle<U> {
+    /// Recasts this handle as a weak handle of an Asset `U`.
+    pub fn cast<U: Asset>(&self) -> Handle<U> {
```

Comment:
I do think `cast` is a better name, but I think we should still make it clear that this is a weak handle in the function name. Maybe `cast_weak`?